igsr_archive/api.py

```diff
@@ -95,10 +96,14 @@ def retrieve_object(self, fireOid=None, firePath=None, outfile=None):
             api_logger.info('Retrieving a FIRE object through its FIRE path')
             url = f"{endpoint}" \
                   f"{firePath}"
-            
-            
-        result = subprocess.run(['aws', 's3', 'cp', url, outfile, '--no-sign-request', '--endpoint-url', endpoint_url], capture_output=True)
         
+        #before using aws, check that aws is available in the environment 
+        if  os.environ.get('__LMOD_REF_COUNT_LOADEDMODULES') is None or "awscli-1.16.308-gcc-9.3.0-47v4b2x" not in os.environ['__LMOD_REF_COUNT_LOADEDMODULES']:
```

Comment:
The check can be more generic as `aws` may be available to the environment even without env modules. Works as expected.